CONTRIBUTING.md

```diff
@@ -68,6 +68,12 @@ follow work through the issue/pull request process.
 5. **Open a pull request into `develop`.** Our main branch is `develop`, and
    this is where we want new work to be merged into. `master` is a release
    branch only.
+6. **Update `CHANGELOG.md` in your branch.** Add the pull-request number and 
+   link, and a brief, one line description under the next tag title. Make sure 
```

Comment:
I still think it should be added. Without it we would need to be a lot more descriptive about each change, specially breaking changes. With the PR link it becomes easy to look up exactly which lines in which files were changed. Without it the only indication about what was done and how to adapt to it would have to be de description, and I would like to keep it short, one liners.

>[...] the changelog notes won’t belong to the actual commit of the change

I don't think they should. I would, one way or another, add it as a dedicated commit, the last on the PR, like: `[refs #xxxx] add changelog entry`. Also, it's pretty uncommon for a PR to be reviewed right after it's submitted, so I don't think that little delay would be a problem. You are right about everything else, but I think it's worth it, no? Or is there a easier way to point the reader to the exact changes, in a easy to visualize way?